src/config/ConfigBuilder.ts

```diff
@@ -345,14 +359,72 @@ export class ConfigBuilder {
         }
     }
 
+    private handleGlobalSerializer(globalSerializer: any) {
+        if (typeof globalSerializer.id !== 'number'
+            || typeof globalSerializer.read !== 'function' || typeof globalSerializer.write !== 'function') {
+            throw new RangeError(
+                `Invalid global serializer given: ${globalSerializer}. Check out the API documentation for the expected object.`
+            );
+        }
+        this.effectiveConfig.serialization.globalSerializer = globalSerializer;
+    }
+
+    private handlePortableFactories(portableFactories: any) {
+        if (typeof portableFactories !== 'object') {
+            throw new RangeError(`Expected 'portableFactories' to be an object but it is a: ${typeof portableFactories}`);
+        }
```

Comment:
I added this `typeof === 'object'` check to all config options which explicitly mention an object in the type definition. Like:

`nearCaches?: { [name: string]: NearCacheConfig };`